### DISCUSSION TITLE
NavBar Not Working in Some Responsive Contexts

### DISCUSSION THREAD

**Discussion by leimao** [ppoffice/hexo-theme-icarus#1048] [Q&A] [answered]
On some Tablets, the NavBar overflow-x and right padding do not work. Do you have any suggestion to fix this? Thank you.

![overflow-x](https://user-images.githubusercontent.com/17606112/158042216-fa4bc199-fdb8-4255-830b-58be09423822.png)

**Comment by leimao** [comment]
A trick to work around is to change this [line](https://github.com/ppoffice/hexo-theme-icarus/blob/3b14e379876386b4da077ddb6cc0d0e57b64b637/layout/common/navbar.jsx#L43) to

```html
<div class="container" style="overflow-x: auto;">
```

@ppoffice Please let me know if there is any more elegant solution and if I should file a merge request for it. Thank you.

**Reply by ppoffice** [reply]
It is not designed to fit too many menu items.

**Reply by leimao** [reply]
This, however, introduces a side-effect.

![image](https://user-images.githubusercontent.com/17606112/158042931-9e11f6ea-ef6f-4e66-a81d-28fc8a34e45a.png)

In a desktop browser, the scroll appears and it does not look good.

**Comment by leimao** [comment, marked as answer]
@ppoffice Created a merge request  #1049 for fixing this issue. Feel free to merge it.
Now the website looks fine in any responsive context:
https://leimao.github.io/
